runtime/tr.source/trj9/env/VMJ9.cpp

```diff
@@ -7473,6 +7473,72 @@ TR_J9VMBase::releaseCodeEstimator(TR::Compilation *comp, TR_EstimateCodeSize *es
    comp->allocator().deallocate(estimator, sizeof(TR_J9EstimateCodeSize));
    }
 
+void
+TR_J9VM::transformJavaLangClassIsArray(TR::Compilation * comp, TR::Node * callNode, TR::TreeTop * treeTop)
+   {
+   // Example for the transformation
+   // treetop (may be null check)
+   //   icalli                   <= callNode
+   //     aload <parm 1>         <= jlClass
+   //
+   //
+   // Final: (when target.is32Bit() == true)
+   //
+   // NULLCHK (if there is a null check)
+   //   PassThrough
+   //     aload <parm 1>         <= jlClass
+   // treetop
+   //   idiv
```

Comment:
I don't follow the idiv here... could you explain a bit more?

Reply:
Class.isArray returns boolean so the result can only be 0 or 1.

Reply:
should this be a right shift or something if we are trying to get rid of trailing zeros or something then? idiv is an expensive thing that the optimizer needs to see through so we should avoid generating it

Reply:
Since the divisor is a constant, it will be turned into a shift instruction in codegen.

Reply:
Since you're creating the expression yourself I suggest you create the optimal one now rather than relying on the code generator to do any sort of optimization on it. We have multiple code generators and creating trees that then need to be optimized in all of them is not such a good idea.

Reply:
Make sense, will make the change

Reply:
idiv has been replaced with iushr.